src/main/kotlin/lotto/domain/WinningLotto.kt

```diff
@@ -4,18 +4,34 @@ class WinningLotto(
     val winningNumbers: Lotto,
     val bonusNumber: LottoNumber,
 ) {
-    fun matchLotto(targetLotto: Lotto): Rank {
+    init {
+        validateDuplicate()
+    }
+
+    fun extractWinningResult(order: Order): WinningResult {
+        val lottoRanks =
+            (order.autoLottos + order.manualLottos).map { this.matchLotto(it) }
+                .groupingBy { it }
+                .eachCount()
+        return WinningResult.from(lottoRanks, order.amount)
```

Comment:
```suggestion
        val lottoRanks =  (order.autoLottos + order.manualLottos).map { this.matchLotto(it) }
        return WinningResult.from(lottoRanks, order.amount)
```
groupingBy 조차도 분석에 해당하는 로직은 아닐까요?
WinningResult에는 당첨정보만 있는 구조는 어떨까요?